Validation/CTPPS/python/simu_config/year_2016_cff.py

```diff
@@ -9,14 +9,16 @@
   )
 )
 
-# geometry
-from Geometry.VeryForwardGeometry.geometryRPFromDD_2017_cfi import * # using 2017 here is OK
-ctppsCompositeESSource.compactViewTag = ctppsGeometryESModule.compactViewTag
-del ctppsGeometryESModule # this functionality is replaced by the composite ES source
+# geometry (using 2017 here is OK)
+from Geometry.VeryForwardGeometry.geometryRPFromDD_2017_cfi import totemGeomXMLFiles, ctppsDiamondGeomXMLFiles, ctppsUFSDGeomXMLFiles, ctppsPixelGeomXMLFiles
+from Geometry.VeryForwardGeometry.geometryRPFromDD_2017_cfi import XMLIdealGeometryESSource_CTPPS
+from Geometry.VeryForwardGeometry.geometryRPFromDD_2017_cfi import ctppsGeometryESModule as _geom
+ctppsCompositeESSource.compactViewTag = _geom.compactViewTag
+ctppsCompositeESSource.isRun2 = _geom.isRun2
```

Comment:
Is this configuration fragment used in production or only in private tests? (by quick look it looks like the latter)

I'm asking because these lines modify a python object "owned" by (defined in) another python fragment. It may lead to surprising "long-distance interaction", like in this pathological example
```py
process.load("Path.To.fileThatDefines_ctppsCompositeESSource_cff")
import Path.To.fileThatDefines_ctppsCompositeESSource_cff
# process.ctppsCompositeESSource.isRun2 is now set as in year_2016_cff
```
Therefore in production configurations a python object is allowed to be modified only in the file that defines it.

Written that, for private test stuff where you have full control of everything, this could be ok, but be aware that the setup is somewhat fragile.

Reply:
@jan-kaspar can confirm, but these are only used to define validation objects used in unit tests (hence these last minute modifications after the merge of #32207).

Reply:
The only "central" use is the unit test, as @forthommel writes.

Then, we (will) use these files extensively in PPS for "private" forward-proton simulations (cf. https://twiki.cern.ch/twiki/bin/viewauth/CMS/TaggedProtonsDirectSimulation). But I guess that this falls in the category "we have full control of everything".